helm/nginx-ingress-controller-app/values.yaml

```diff
@@ -87,6 +86,13 @@ controller:
       http: 30010
       https: 30011
 
+  metrics:
+    enabled: false
```

Comment:
@rossf7 @pipo02mix I defined this flag here and not in configmap starting above on line 24 (comments on 22); was that correct? IIUC that configmap above is for tuning nginx itself with nginx specific config options.

Reply:
Yes this is correct. The configmap is for the ingress controller itself. It's templated based on the chart values.

Reply:
if this is `false`, why we need to specify ports anyway?

Reply:
@tomahawk28 to have defaults ready so it's easy to enable metrics service with simple custom `values.yaml` e.g. during app installation
```
controller:
  metrics:
    enabled: true
```

Reply:
@rossf7 what confused me was this comment for configmap
https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/helm/nginx-ingress-controller-app/values.yaml#L22-L23
stating

> configmap contains settings that can be overridden with a custom values configmap

which is true to some extent but it's not entirely clear to which configmap it refers to (our nginx-ingress-controller-user-values on CP or upstream documented ingress-nginx ConfigMap); any property in values.yaml like now metrics service enabling can be overriden with nginx-ingress-controller-user-values ConfigMap, a custom values configmap.

I believe that comment would be more accurate that that configmap is for configuring upstream documented nginx configmap directives (so ingress-nginx ConfigMap) https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#configuration-options

Another source of confusion are hpa settings placed under configmap there https://github.com/giantswarm/nginx-ingress-controller-app/blob/master/helm/nginx-ingress-controller-app/values.yaml#L48-L53 which are not controller's configmap directives. Those HPA settings should be placed outside of configmap property. In upstream chart they have HPA controlled via .Values.controller.autoscaling.* flags

Will remember to fix this in future iterations.

Reply:
@sslavic Sorry yes that comment is confusing and it has a very specific meaning.

For chartconfig CRs we were worried about users changing things. So we took a whitelist approach and only allowed values under `.Values.configmap` to be changed.

The approach was too restrictive and we removed it for app CRs. So any value can be overridden. When we migrated this chart to an app CR we kept the values structure and this comment.

I think the comment should be updated so its meaning is clear. Also over time we should improve the values structure since this restriction no longer applies.

Reply:
thanks for heads up @rossf7!